### PR TITLE
CP-33380: Use dune 2.0

### PR DESCRIPTION
**Pull request by psafont**
No major problems found when building the toolstack on a local opam environment.

The biggest caveat is that sha is now pinned to the latest commit of https://github.com/djs55/ocaml-sha/pull/46

Aside from that I updated ppx_deriving_rpc to the same version as the rest of rpclib packages and updates stdext to reflect upstream packages

### CODE REVIEW

File: packages/upstream/dune-private-libs.2.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,42 @@
+opam-version: "2.0"
+synopsis: "Private libraries of Dune"
+description: """
```
Comment (lippirk) [inline packages/upstream/dune-private-libs.2.4.0/opam]: :thinking:

**Comment by lippirk** [comment]
LGTM - it's nice that we didn't have to update base  :)

**Comment by psafont** [comment]
crowbar needs to be updated in a similar way to  sha, there might be more libraries in upstream-extra as well.

**Comment by psafont** [comment]
travis is failing when compiling base, strange it didn't fail locally

**Comment by lindig** [comment]
```

**Comment by psafont** [comment]
That's due to a missing dependency in the opam metadata.

**Comment by lindig** [comment]
```
#=== ERROR while compiling message-switch.master ==============================#
1364# context              2.0.6 | linux/x86_64 | ocaml-base-compiler.4.08.1 | file:///mnt
1365# path                 ~/.opam/4.08/.opam-switch/build/message-switch.master
1366# command              ~/.opam/4.08/bin/dune runtest -p message-switch -j 71
1367# exit-code            1
1368# env-file             ~/.opam/log/message-switch-1059-8a3c98.env
1369# output-file          ~/.opam/log/message-switch-1059-8a3c98.out
1370### output ###
1371# File "_build/.dune/default/core_test/dune", line 13, characters 0-362:
1372# 13 | (alias
1373# 14 |   (name runtest)
1374# 15 |   (deps
1375# ....
1376# 27 |   (package message-switch)
1377# 28 |   (action (run ./basic-rpc-test.sh))
1378# 29 | )
1379# Error: No rule found for cli/main.exe
```

**Comment by lindig** [comment]
The failure is in the build with the tests, the one without passes.

**Comment by lindig** [comment]
Looks like it's good to go

**Pull request merged**